### PR TITLE
WIP: Adding C api for the NexusPlot class

### PR DESCRIPTION
**Pull request by JensGM**

### CODE REVIEW

**Comment by joakim-hove** [comment]
Looks good - thank you :-)

File: lib/nexus/nexus_plot_c.cpp [hunk 1]
```diff
@@ -0,0 +1,91 @@
+/*
+   Copyright (C) 2017 Statoil ASA, Norway.
+
+   The file 'nexus_plot.hpp' is part of ERT - Ensemble based Reservoir Tool.
```
Comment (andreabrambilla) [inline lib/nexus/nexus_plot_c.cpp]: file name should be probably fixed

File: lib/nexus/tests/nexus_plot_c_api.c [hunk 2]
```diff
@@ -0,0 +1,60 @@
+/*
+   Copyright (C) 2017 Statoil ASA, Norway.
+   The file 'nexus_assert_null.c' is part of ERT - Ensemble based Reservoir Tool.
```
Comment (andreabrambilla) [inline lib/nexus/tests/nexus_plot_c_api.c]: file name

**Comment by andreabrambilla** [comment]
@joakim-hove @JensGM Can I ask what's the point of having C bindings for the nexus stuff? I mean, the nexus stuff uses C++ (11), so we need a compiler supporting C++ to use it, so why not writing C++ code when using it? 

Ok, I understand that this would allow to use the nexus stuff in libres without requiring a C++ compiler for libres... but I'd rather use c++ code also in libres and put it behind some proper cmake flags (similar to https://github.com/Statoil/libecl/pull/205)

**Comment by JensGM** [comment]
@joakim-hove Alright, no problem

File: lib/nexus/nexus_plot_c.cpp [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+        UTIL_TYPE_ID_INIT( plt, NEXUS_PLOT_ID );
+        plt->data = nexus_plt;
+        return plt;
+    }
+    catch ( std::runtime_error& e ) {
+        return NULL;
+    }
+}
+void nexus_plot_free( nexus_plot_type *plt ) {
+    delete plt->data;
```
[the remark this answers is not in the document]
Comment (andreabrambilla) [reply lib/nexus/nexus_plot_c.cpp]: I guess this refers to line 53, not to line 52

**Pull request closed without merging**